src/vcpkg/postbuildlint.cpp

```diff
@@ -730,6 +730,47 @@ namespace vcpkg::PostBuildLint
         return LintStatus::SUCCESS;
     }
 
+    static LintStatus check_pkgconfig_dir_only_in_lib_dir(const Filesystem& fs, const Path& dir)
+    {
+        const auto allowed_in_share = [&fs](const Path& path) {
+            // If there is a "Lib:" or "Libs.private:" line in the .pc file, then the file must be in /lib/pkgconfig
+            return !Util::any_of(fs.read_lines(path, VCPKG_LINE_INFO),
+                                 [](const auto& line) { return Strings::starts_with(line, "Libs"); });
+        };
+
+        std::vector<Path> pkgconfig_files = fs.get_regular_files_recursive(dir, IgnoreErrors{});
+        Util::erase_remove_if(pkgconfig_files, [&allowed_in_share](const Path& path) {
+            if (!Strings::ends_with(path, ".pc")) return true;
+            const auto parent_path = Path(path.parent_path());
+            if (parent_path.filename() != "pkgconfig") return false;
+            const auto pkgconfig_parent_name = Path(parent_path.parent_path()).filename().to_string();
+            return pkgconfig_parent_name == "lib" || pkgconfig_parent_name == "manual-link" ||
+                   (pkgconfig_parent_name == "share" && allowed_in_share(path));
+        });
+
+        if (!pkgconfig_files.empty())
+        {
+            print2(Color::warning, "There should be no pkgconfig directories outside of lib and debug/lib.\n");
+            print2("The following misplaced pkgconfig files were found:\n");
+            print_paths(pkgconfig_files);
+            print2(
+                Color::warning,
+                "You can move the pkgconfig files with the following commands:\n"
+                "\n"
+                R"###(    file(INSTALL "${CURRENT_PACKAGES_DIR}/a/path/pkgconfig/name.pc" DESTINATION "${CURRENT_PACKAGES_DIR}/a/path/pkgconfig"))###"
+                "\n"
+                R"###(    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/a/path/pkgconfig"))###"
```

Comment:
Should we emit the actual names we detected in this replacement text?

Reply:
But we don't know the correct destination path.

Reply:
You at least know the source one right? And we can always recommend `lib/pkgconfig/original-name.pc` as a good first guess, right? (I'm genuinely asking about the second question, I'm not as familiar with pkgconfig as I should be.)

Reply:
Yeah we know the source, but maybe something like `file(RENAME old_pkg_config_dir <newname>)` is a better solution in general. And there are some libs with a lot of .pc files (ca. 350)

Reply:
I think this initial message is fine for now, and if we get confusion in the issues, we can figure out a better way to say this.